### PR TITLE
[docs][denylist] Updated the create coin topic to include reg coins

### PR DESCRIPTION
**Pull request by ronny-mysten**
## Description 

Updated the create coin topic to include reg coins

## Test Plan 

👀 

---
If your changes are not user-facing and do not break anything, you can skip the following section. Otherwise, please briefly describe what has changed under the Release Notes section.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: docs/content/guides/developer/sui-101/create-coin.mdx [hunk 1]
```diff
@@ -4,26 +4,177 @@ title: Create a Coin
 
 Publishing a coin on Sui is nearly as straightforward as publishing a new type. The main difference is the requirement of a [one-time witness](/concepts/sui-move-concepts/one-time-witness.mdx) when creating a coin.
 
-```move
+```rust
```
Comment (tnowacki) [inline docs/content/guides/developer/sui-101/create-coin.mdx:7]: Why the change here? I thought we had move highlighting? Or do we not? :(

File: docs/content/guides/developer/sui-101/create-coin.mdx [hunk 2]
```diff
@@ -4,26 +4,177 @@ title: Create a Coin
...
-    use sui::coin;
+    use sui::coin::{Self, Coin, TreasuryCap};
     use sui::transfer;
     use sui::tx_context::{Self, TxContext};
 
     /// The type identifier of coin. The coin will have a type
     /// tag of kind: `Coin<package_object::mycoin::MYCOIN>`
     /// Make sure that the name of the type matches the module's name.
-    struct MYCOIN has drop {}
+    struct MYCOIN has drop {};
```
Comment (tnowacki) [inline docs/content/guides/developer/sui-101/create-coin.mdx:17]: Will need to revert this one
```suggestion
    struct MYCOIN has drop {}
```

File: docs/content/guides/developer/sui-101/create-coin.mdx [hunk 3]
```diff
@@ -4,26 +4,177 @@ title: Create a Coin
...
 
     /// Module initializer is called once on module publish. A treasury
     /// cap is sent to the publisher, who then controls minting and burning
     fun init(witness: MYCOIN, ctx: &mut TxContext) {
         let (treasury, metadata) = coin::create_currency(witness, 6, b"MYCOIN", b"", b"", option::none(), ctx);
         transfer::public_freeze_object(metadata);
-        transfer::public_transfer(treasury, tx_context::sender(ctx))
+        transfer::public_transfer(treasury, tx_context::sender(ctx));
```
Comment (tnowacki) [inline docs/content/guides/developer/sui-101/create-coin.mdx:24]: Technically don't need this `;`, though I suppose its fine

File: docs/content/guides/developer/sui-101/create-coin.mdx [hunk 4]
```diff
@@ -4,26 +4,177 @@ title: Create a Coin
...
     }
 }
 ```
 
 The `Coin<T>` is a generic implementation of a coin on Sui. The owner of the `TreasuryCap` gets control over the minting and burning of coins. Further transactions can be sent directly to the `sui::coin::Coin` with `TreasuryCap` object as authorization.
+
+Extending the example further, add a `mint` function to the module. Use the `mint_and_transfer` function of the `Coin` module to create (mint) a coin and transfer it to an address.
```
Comment (tnowacki) [inline docs/content/guides/developer/sui-101/create-coin.mdx:31]: I feel like we don't really advocate for using mint_and_transfer much anymore? I feel its mostly there for historical reasons, though maybe @manolisliolios or @damirka have thoughts

File: docs/content/guides/developer/sui-101/create-coin.mdx [hunk 5]
```diff
@@ -4,26 +4,177 @@ title: Create a Coin
...
 ```
 
 The `Coin<T>` is a generic implementation of a coin on Sui. The owner of the `TreasuryCap` gets control over the minting and burning of coins. Further transactions can be sent directly to the `sui::coin::Coin` with `TreasuryCap` object as authorization.
+
+Extending the example further, add a `mint` function to the module. Use the `mint_and_transfer` function of the `Coin` module to create (mint) a coin and transfer it to an address. 
+
+```rust
+public fun mint(
+        treasury_cap: &mut TreasuryCap<MYCOIN>, amount: u64, recipient: address, ctx: &mut TxContext
+    ) {
```
Comment (tnowacki) [inline docs/content/guides/developer/sui-101/create-coin.mdx:36]:
```suggestion
    treasury_cap: &mut TreasuryCap<MYCOIN>, 
    amount: u64, 
    recipient: address, 
    ctx: &mut TxContext,
) {
```

File: docs/content/guides/developer/sui-101/create-coin.mdx [hunk 6]
```diff
@@ -4,26 +4,177 @@ title: Create a Coin
...
+        coin::mint_and_transfer(treasury_cap, amount, recipient, ctx)
+    }
```
Comment (tnowacki) [inline docs/content/guides/developer/sui-101/create-coin.mdx:38]:
```suggestion
    coin::mint_and_transfer(treasury_cap, amount, recipient, ctx)
}
```

File: docs/content/guides/developer/sui-101/create-coin.mdx [hunk 7]
```diff
@@ -4,26 +4,177 @@ title: Create a Coin
...
+
+## DenyList
+
+:::info
+
+The `DenyList` object and regulated coins are currently available in Devnet only. 
+
+:::
+
+The Sui framework provides a `DenyList` singleton, shared object that the bearer of a `DenyCap` can use to specify a list of addresses that are unable to use the specified object. This is useful, for example, when creating a regulated coin on Sui that requires the ability to block certain addresses from using it as inputs to transactions. Regulated coins on Sui satisfy any regulations that require the ability to prevent known bad actors from having access to those coins.
```
Comment (tnowacki) [inline docs/content/guides/developer/sui-101/create-coin.mdx:57]:
```suggestion
The Sui framework provides a `DenyList` singleton, shared object that the bearer of a `DenyCap` can use to specify a list of addresses that are unable to use coins of the specified type. This is useful, for example, when creating a regulated coin on Sui that requires the ability to block certain addresses from using it as inputs to transactions. Regulated coins on Sui satisfy any regulations that require the ability to prevent known bad actors from having access to those coins. 
```
> that are unable to use coins of the specified type

I'm struggling with this part here that I edited because in general the DenyList can be used for any core type defined in `sui`, but right now it's limited to coins

File: docs/content/guides/developer/sui-101/create-coin.mdx [hunk 8]
```diff
@@ -4,26 +4,177 @@ title: Create a Coin
...
+
+The `DenyList` object is a system object that has the address `0x403`. You cannot create it yourself.
+
+:::
+
+## Create regulated coin
+
+If you need the ability to deny specific addresses from having access to your coin, you can use the `create_regulated_currency` function (instead of `create_currency`) to create it.
+
+Behind the scenes, `create_regulated_currency` uses the `create_currency` function to create the coin, but produces a `DenyCap` object that allows its bearer to control access to the coin through a `DenyList` object. Consequently, the way to create a coin using `create_regulated_currency` is similar to the previous example, with the addition of a transfer of the `DenyList` object to the module publisher.
```
Comment (tnowacki) [inline docs/content/guides/developer/sui-101/create-coin.mdx:69]:
```suggestion
Behind the scenes, `create_regulated_currency` uses the `create_currency` function to create the coin, but also produces a `DenyCap` object that allows its bearer to control access to the coin's deny list in a `DenyList` object. Consequently, the way to create a coin using `create_regulated_currency` is similar to the previous example, with the addition of a transfer of the `DenyCap` object to the module publisher.
```

File: docs/content/guides/developer/sui-101/create-coin.mdx [hunk 9]
```diff
@@ -4,26 +4,177 @@ title: Create a Coin
...
+module examples::regcoin {
+    use std::option;
+    use sui::coin;
+    use sui::transfer;
+    use sui::tx_context::{Self, TxContext};
+
+    struct REGCOIN has drop {}
+
+    fun init(witness: REGCOIN, ctx: &mut TxContext) {
+        let (treasury, denylist, metadata) = coin::create_regulated_currency(witness, 6, b"REGCOIN", b"", b"", option::none(), ctx);
```
Comment (tnowacki) [inline docs/content/guides/developer/sui-101/create-coin.mdx:81]:
```suggestion
        let (treasury, deny_cap, metadata) = coin::create_regulated_currency(witness, 6, b"REGCOIN", b"", b"", option::none(), ctx);
```

File: docs/content/guides/developer/sui-101/create-coin.mdx [hunk 10]
```diff
@@ -4,26 +4,177 @@ title: Create a Coin
...
+        transfer::public_freeze_object(metadata);
+        transfer::public_transfer(treasury, tx_context::sender(ctx));
+        transfer::public_transfer(denylist, tx_context::sender(ctx))
```
Comment (tnowacki) [inline docs/content/guides/developer/sui-101/create-coin.mdx:84]:
```suggestion
        transfer::public_transfer(deny_cap, tx_context::sender(ctx))
```

File: docs/content/guides/developer/sui-101/create-coin.mdx (hunk 1 above)
Comment (ronny-mysten) [reply docs/content/guides/developer/sui-101/create-coin.mdx:7]: We used to not. Looks like the SF team added, so thanks for pointing out.

**Review by lxfind — APPROVED** [review]
Unblock
